include/extractor/guidance/intersection.hpp

```diff
@@ -64,25 +75,24 @@ struct Intersection final : public std::vector<ConnectedRoad>
 {
     using Base = std::vector<ConnectedRoad>;
 
-    inline Base::iterator findClosestTurn(double angle)
-    {
-        return std::min_element(this->begin(),
-                                this->end(),
-                                [angle](const ConnectedRoad &lhs, const ConnectedRoad &rhs) {
-                                    return util::guidance::angularDeviation(lhs.turn.angle, angle) <
-                                           util::guidance::angularDeviation(rhs.turn.angle, angle);
-                                });
-    }
-
-    inline Base::const_iterator findClosestTurn(double angle) const
-    {
-        return std::min_element(this->begin(),
-                                this->end(),
-                                [angle](const ConnectedRoad &lhs, const ConnectedRoad &rhs) {
-                                    return util::guidance::angularDeviation(lhs.turn.angle, angle) <
-                                           util::guidance::angularDeviation(rhs.turn.angle, angle);
-                                });
-    }
+    /*
+     * find the turn whose angle offers the least angularDeviation to the specified angle
+     * E.g. for turn angles [0,90,260] and a query of 180 we return the 260 degree turn (difference
+     * 80 over the difference of 90 to the 90 degree turn)
+     */
+    Base::iterator findClosestTurn(double angle);
+    Base::const_iterator findClosestTurn(double angle) const;
+
+    /*
+     * Check validity of the intersection object. We assume a few basic properties every set of
+     * connected roads should follow throughout guidance pre-processing. This utility function
+     * allows checking intersections for validity
+     */
+    bool valid() const;
+
+    // given all possible turns, which is the highest connected number of lanes per turn. This value
+    // is used, for example, during generation of intersections.
+    std::uint8_t getHighestConnectedLaneCount(const util::NodeBasedDynamicGraph &) const;
```

Comment:
What about functionality for checking / getting the UTurn ConnectedRoad?

And / Or all roads except the UTurn road? Maybe

    auto ThinkAboutAGoodNameForNonUTurnRoadsBegin() { return begin() + 1; }
    auto ThinkAboutAGoodNameForNonUTurnRoadsEnd() { return end; }

with assert on !empty, const overloads, make_iterator_range overloads.